src/js/modules/supervisors/FileManager.ts

```diff
@@ -140,21 +145,66 @@ class FileManager {
    * @param repository, is a coprocessor container
    */
   updateRepositoryOnNewFile(repository: Repository): void {
-    return this.watcher.on("add", (filePath) => {
+    return this.submitDirWatcher.on("add", (filePath) => {
       this.addCoprocessor(filePath, repository).catch((e) =>
         console.error(e.message)
       );
       //TODO: implement winston for logging information and error handler
     });
   }
 
+  /**
+   * add event listener for removing event from active folder
+   * @param repository
+   */
+  removeHandleOnDeleteFile(repository: Repository): Promise<void> {
+    return this.activeDirWatcher.on("unlink", (filePath) =>
+      this.removeHandleFromFilePath(filePath, repository)
+    );
+  }
+
+  /**
+   * Updates the given Repository instance when a coprocessor is removed
+   * from folder.
+   * @param filePath, removed handle path
+   * @param repository, is a coprocessor container
+   */
+  removeHandleFromFilePath(
+    filePath: string,
+    repository: Repository
+  ): Promise<void> {
+    const name = path.basename(filePath, ".js");
+    const id = hash64(Buffer.from(name), 0).readBigUInt64LE();
+    const [repositoryHandle] = repository.getHandlesByCoprocessorIds([id]);
+    if (!repositoryHandle) {
+      console.error(
+        "error: trying to disable a removed coprocessor from " +
+          "'active' folder but it wasn't loaded in memory, file name " +
+          `${name}.js`
+      );
+      return Promise.resolve();
+    } else {
+      return this.disableCoprocessors([repositoryHandle.coprocessor])
+        .then(() => this.repository.remove(repositoryHandle))
+        .then(() =>
+          console.log(
+            `disabled coprocessor: ID ` +
+              repositoryHandle.coprocessor.globalId +
+              " filename: " +
+              `${name}.js `
```

Comment:
There's an unnecessary trailing space after "js". Also I think this and the previous line could go in the same line as a template string.

Reply:
Done